src/utils/circular_array_buffer.jl

```diff
@@ -165,12 +153,12 @@ end
 `update!` the last frame of `cb` with data.
 """
 function RLBase.update!(cb::CircularArrayBuffer{T,N}, data::AbstractArray) where {T,N}
-    select_last_dim(cb.buffer, _buffer_frame(cb, cb.length)) .= data
+    select_last_dim(cb.buffer, _buffer_frame(cb, cb.nframes)) .= data
     cb
 end
 
-function RLBase.update!(cb::CircularArrayBuffer{T,1}, data) where {T}
-    cb.buffer[_buffer_frame(cb, cb.length)] = data
+function RLBase.update!(cb::CircularArrayBuffer{T,1}, data::T) where {T}
```

Comment:
The type constraint is not necessary here. Sometimes the eltype in `cb` is `Float32`, and we'd like the data of type `Float64` or `Int` to be pushed into it instead of throwing an `MethodNotfFound` error.

Reply:
Better to add a test case for the scenario I described above.

Reply:
Ah.. I see. Yes, this is better.
By the way,
>Sometimes the eltype in `cb` is `Float32`, and we'd like the data of type `Float64` or `Int` to be pushed into it instead of throwing an `MethodNotfFound` error.

Did you mean eltype of `cb` would be `Float64` and we should allow pushing `Float32` as well?
Since not all `Float64` can be represented by a `Float32`, right?

Reply:
A data of type `Float64` will be truncated when converting to `Float32`.

Reply:
Oh, Okay. I didn't know that. Does it always happen? I thought Julia would not allow this conversion to happen always. For example, in the cases when trucation leads to a loss of information.

Reply:
Yes I think so.